internal/cfgplugins/policyforwarding.go

```diff
@@ -399,3 +400,48 @@ func PushPolicyForwardingConfig(t *testing.T, dut *ondatra.DUTDevice, ni *oc.Net
 	niPath := gnmi.OC().NetworkInstance(ni.GetName()).Config()
 	gnmi.Replace(t, dut, niPath, ni)
 }
+
+// Configure GRE decapsulated. Adding deviation when device doesn't support OC
+func NewConfigureGRETunnel(t *testing.T, dut *ondatra.DUTDevice, decapIp string, policyName string, portName string) {
+	d := &oc.Root{}
+	ni1 := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut))
+	ni1.SetType(oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+	npf := ni1.GetOrCreatePolicyForwarding()
+	np := npf.GetOrCreatePolicy(policyName)
+	np.PolicyId = ygot.String(policyName)
+	np.Type = oc.Policy_Type_PBR_POLICY
+
+	npRule := np.GetOrCreateRule(10)
+	ip := npRule.GetOrCreateIpv4()
+	ip.DestinationAddressPrefixSet = ygot.String(decapIp)
+	npAction := npRule.GetOrCreateAction()
+	npAction.DecapsulateGre = ygot.Bool(true)
+
+	port := dut.Port(t, portName)
+	ingressPort := port.Name()
+	t.Logf("Applying forwarding policy on interface %v ... ", ingressPort)
+
+	intf := npf.GetOrCreateInterface(ingressPort)
+	intf.ApplyForwardingPolicy = ygot.String(policyName)
+	intf.GetOrCreateInterfaceRef().Interface = ygot.String(ingressPort)
+
+	gnmi.Update(t, dut, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Config(), ni1)
```

Comment:
Same here,  this should use gnmi.BatchReplace

Reply:
Added changes